src/util/bip143.rs

```diff
@@ -102,9 +104,9 @@ impl SighashComponents {
 }
 
 /// A replacement for SigHashComponents which supports all sighash modes
-pub struct SigHashCache<'a> {
+pub struct SigHashCache<R: Deref<Target=Transaction>> {
```

Comment:
I was initially concerned if this is an unidiomatic abuse of `Deref`, but after some thought I think it's fine. `SigHashCache` works with any smart pointer to `Transaction` which is what `Deref` is for.